examples/TimeTBB.cpp

```diff
@@ -153,7 +154,6 @@ int main(int argc, char* argv[])
   for(size_t n: numThreads)
   {
     cout << "With " << n << " threads:" << endl;
-    tbb::task_scheduler_init init((int)n);
```

Comment:
Without this line this entire timing script is pointless. Is there not a replacement for this deprecated function? Otherwise, could we just suppress the deprecation warning?

Reply:
Why do you say that? The `task_scheduler_init` was optional in the first place. If we don't provide one, TBB uses one internally. We are still timing how fast the parallelization occurs. From what I can tell the functionality of the script has not changed one bit.
Suppressing the deprecation warning is just hiding the problem. In future releases this will become an error, instead of a warning.

Reply:
I do understand that it may be hard to support both TBB 4.4 and TBB 2020 and if that is the case, we can just add a requirement during the installation procedure that warns the use of TBB > 2020. Although I think we can get GTSAM to support both versions.

Reply:
No. Read the for loop more carefully. The purpose of this script is to time the performance of TBB with different grain sizes, and this default grain size varies with number of threads. In order to read this default grain size, this loop initializes tbb with 1, 4, 8 threads, and reads the grain size.
Removing `tbb::task_scheduler_init init((int)n);` causes number of threads to be equal to number of cores for all cases, rendering the for loops in this test redundant and pointless.

Reply:
Ahh I see now, yes we are specifying the number of threads each iteration of the for loop. Hmmm, well this is a predicament.

Reply:
Let me link the TBB documentation: https://software.intel.com/en-us/node/506296 regarding `task_scheduler_init`

Reply:
I believe we can use `task_arena` instead. (https://software.intel.com/en-us/node/506359)

Reply:
I'm not sure on how to replicate this script without `task_scheduler_init` so I am going to leave this script be, hopefully it will get tackled before the feature gets removed.